File: infra/solutions/cost_effective_scalable_inference_agentic_ai_automode/agentic-apps/agentic_rag_opensearch/src/test-agents.ts
import 'dotenv/config';
import { KnowledgeAgent, RAGAgent, MCPAgent, SupervisorAgent } from "./agents";
import MCPClient from "./MCPClient";
import { logTitle } from "./utils";
import path from "path";

// Simple test to verify all agents can be initialized
async function testAgents() {
    logTitle('TESTING MULTI-AGENT SYSTEM');
    
    try {
        // Test Knowledge Agent
        console.log('Testing Knowledge Agent...');
        const knowledgeAgent = new KnowledgeAgent();
        await knowledgeAgent.init();
        const hasChanges = await knowledgeAgent.checkForChanges();
        console.log(`Knowledge changes detected: ${hasChanges}`);
        await knowledgeAgent.close();
        console.log('✓ Knowledge Agent test passed');
        
        // Test RAG Agent
        console.log('\nTesting RAG Agent...');
        const ragAgent = new RAGAgent();
        await ragAgent.init();
        const stats = ragAgent.getStats();
        console.log('RAG Agent stats:', stats);
        await ragAgent.close();
        console.log('✓ RAG Agent test passed');
        
        // Test MCP Agent
        console.log('\nTesting MCP Agent...');
        const outPath = path.resolve(process.cwd(), 'output');
        const fileMCP = new MCPClient("filesystem", "npx", ['-y', '@modelcontextprotocol/server-filesystem', outPath]);
        const mcpAgent = new MCPAgent([fileMCP]);
        await mcpAgent.init();
        const mcpStats = mcpAgent.getStats();
        console.log('MCP Agent stats:', mcpStats);
        const tools = await mcpAgent.listAvailableTools();
        console.log(`Available tools: ${tools.length}`);
        await mcpAgent.close();
        console.log('✓ MCP Agent test passed');
        
        // Test Supervisor Agent
        console.log('\nTesting Supervisor Agent...');
        const fileMCP2 = new MCPClient("filesystem", "npx", ['-y', '@modelcontextprotocol/server-filesystem', outPath]);
        const supervisor = new SupervisorAgent([fileMCP2]);
        await supervisor.init();
        console.log('Supervisor Agent initialized successfully');
        await supervisor.close();
        console.log('✓ Supervisor Agent test passed');
        
        console.log('\n🎉 All agent tests passed successfully!');
        
    } catch (error) {
        console.error('❌ Agent test failed:', error);
        process.exit(1);
    }
}

// Run the test
(async () => {
    await testAgents();
})();

File: infra/solutions/cost_effective_scalable_inference_agentic_ai_automode/agentic-apps/strandsdk_agentic_rag_opensearch/src/test_agents.py
#!/usr/bin/env python3
"""
Agent Testing Script

This script tests the multi-agent system with various queries to ensure
all components are working correctly, including Langfuse integration.
"""

import sys
import logging
from typing import List, Dict, Any
from .config import config
from .utils.logging import setup_logging, log_title
from .agents.supervisor_agent import supervisor_agent, supervisor_agent_with_langfuse
from .agents.knowledge_agent import knowledge_agent, knowledge_agent_with_langfuse
from .agents.mcp_agent import mcp_agent, mcp_agent_with_langfuse

def main():
    """Main function for testing agents."""
    # Setup logging
    setup_logging()
    logger = logging.getLogger(__name__)
    
    try:
        # Validate configuration
        config.validate_config()
        
        log_title("AGENT TESTING SCRIPT")
        logger.info("Starting agent testing process")
        
        # Test individual agents first
        test_individual_agents()
        
        # Test supervisor agent
        test_supervisor_agent()
        
        print("\n🎉 All tests completed!")
        
    except KeyboardInterrupt:
        print("\n\nTesting interrupted by user.")
        sys.exit(0)
    except Exception as e:
        logger.error(f"Agent testing failed: {e}")
        print(f"❌ Error: {e}")
        sys.exit(1)

def test_individual_agents():
    """Test individual agents."""
    log_title("INDIVIDUAL AGENT TESTS")
    
    # Test Knowledge Agent
    print("🧠 Testing Knowledge Agent...")
    try:
        if config.is_langfuse_enabled():
            knowledge_response = knowledge_agent_with_langfuse("Please scan the knowledge directory and report what files are available.")
            print(f"✅ Knowledge Agent Response (with Langfuse): {str(knowledge_response)[:200]}...")
        else:
            knowledge_response = knowledge_agent("Please scan the knowledge directory and report what files are available.")
            print(f"✅ Knowledge Agent Response: {str(knowledge_response)[:200]}...")
    except Exception as e:
        print(f"❌ Knowledge Agent failed: {e}")
    
    # Test MCP Agent
    print("\n🔧 Testing MCP Agent...")
    try:
        if config.is_langfuse_enabled():
            mcp_response = mcp_agent_with_langfuse("Please help me understand what tools are available for file operations.")
            print(f"✅ MCP Agent Response (with Langfuse): {str(mcp_response)[:200]}...")
        else:
            mcp_response = mcp_agent("Please help me understand what tools are available for file operations.")
            print(f"✅ MCP Agent Response: {str(mcp_response)[:200]}...")
    except Exception as e:
        print(f"❌ MCP Agent failed: {e}")

def test_supervisor_agent():
    """Test the supervisor agent with various queries."""
    log_title("SUPERVISOR AGENT TESTS")
    
    test_queries = [
        "What is the status of the knowledge base?",
        "Can you help me understand what files are in the knowledge directory?",
        "Please search for information about Bell's palsy if available."
    ]
    
    for i, query in enumerate(test_queries, 1):
        print(f"\n🧪 Test {i}: {query}")
        
        try:
            if config.is_langfuse_enabled():
                response = supervisor_agent_with_langfuse(query)
                print(f"✅ Success (with Langfuse): {str(response)[:300]}...")
            else:
                response = supervisor_agent(query)
                print(f"✅ Success: {str(response)[:300]}...")
                
        except Exception as e:
            print(f"❌ Error: {e}")

if __name__ == "__main__":
    main()
